doc/manuals/name-encoding.md

```diff
@@ -0,0 +1,47 @@
+# Database and collection name encoding and decoding
+
+MongoDB imposes a set of restrictions regarding the characters which cannot be used as part of the databases and collection names, as well as other limitations. These restrictions are detailed at https://docs.mongodb.com/manual/reference/limits/
+
+Since the names of the databases and the collections used by the STH component are composed from the concatenation of the values assigned to the service, service path, entity id, entity type, attribute name and attribute type (depending on the chosen data model) and we do not want the potential users of the STH component to inherit the restrictions imposed by MongoDB (as the database used by the STH component), a codification mechanism is used before conforming the name of the databases and collections.
```

Comment:
Is the attribute type really used in any data model? AFAIK, the collection-per-attribute model only uses the attribute name.